### PR TITLE
MB-3148 Add createMTOAgent endpoint to Prime API

### PR DESCRIPTION
**Pull request by abbyoung**
## Description

* Adds new endpoint to Prime API so MTO Agents can be added after shipment creation.
* Prime can now create RELEASING_AGENT and RECEIVING_AGENT for an mtoShipment with no existing agents.
* If the agent already exists, they'll need to use updateMTOAgent instead.

## Setup

**Tests**

```
// Tests for handler
 go test ./pkg/handlers/primeapi/ --testify.m "TestCreateMTOAgentHandler" -v

// Tests for service object
 go test ./pkg/services/mto_agent --testify.m "TestMTOAgentCreator" -v
```
**Manual Testing**

You can manually test the endpoint by making a `POST` request to `/mto-shipments/:mtoShipmentID/agents` with a similar body:
```
{
    "firstName": "Riley",
    "lastName": "Baker",
    "email": "rileybaker@example.com",
    "phone": "555-555-3020",
    "agentType": "RELEASING_AGENT"
}
```

* The shipment must be associated with an MTO that is available to Prime
* It cannot already have an agent of the same type (although worth testing to confirm you get the correct error!)

## Code Review Verification Steps
* [ ] Request review from a member of a different team.
* [ ] Have the Jira acceptance criteria been met for this change?

## References

* [Jira story](https://dp3.atlassian.net/browse/MB-3148) for this change

### CODE REVIEW

File: pkg/handlers/primeapi/mto_agent_test.go [hunk 1]
```diff
@@ -204,3 +206,170 @@ func (suite *HandlerSuite) TestUpdateMTOAgentHandler() {
...
+		mtoagent.NewMTOAgentCreator(suite.DB(), movetaskorder.NewMoveTaskOrderChecker(suite.DB())),
+	}
+	req := httptest.NewRequest("POST", fmt.Sprintf("/mto-shipments/%s/agents", mtoShipment.ID), nil)
+
+	suite.T().Run("200 - OK response Receiving Agent", func(t *testing.T) {
+		// Under test: 	CreateMTOAgentHandler, MTOAgentCreator
+		// Set up: 		Pass in valid payload for a receiving agent.
+		// Expected:	Handler returns 200 response with payload of new agent.
+
+		payload := payloads.MTOAgent(receivingAgent)
```
Comment (shimonacarvalho) [inline pkg/handlers/primeapi/mto_agent_test.go:254]: Just a tip. I've realized that creating a model, and then converting to payload to use in a test obscures many errors such that we don't catch nil references, and we should avoid doing this. 
We should create the payload directly like this: 

```
		agent := &primemessages.MTOAgent{
			ID:            *badID,
			MtoShipmentID: *badID,
			FirstName:     handlers.FmtString("Mary"),
		}
```

File: pkg/handlers/primeapi/mto_agent_test.go [hunk 2]
```diff
@@ -204,3 +206,170 @@ func (suite *HandlerSuite) TestUpdateMTOAgentHandler() {
...
+		suite.IsType(&mtoshipmentops.CreateMTOAgentOK{}, response)
+
+		// Check Values
+		agentOK := response.(*mtoshipmentops.CreateMTOAgentOK)
+		suite.Equal(agentOK.Payload.MtoShipmentID.String(), receivingAgent.MTOShipmentID.String())
+		suite.Equal(string(agentOK.Payload.AgentType), string(receivingAgent.MTOAgentType)) // wasn't updated, should be original value
+		suite.Equal(agentOK.Payload.FirstName, receivingAgent.FirstName)
+		suite.Equal(agentOK.Payload.LastName, receivingAgent.LastName)
+		suite.Equal(agentOK.Payload.Email, receivingAgent.Email)
+		suite.Equal(agentOK.Payload.Phone, receivingAgent.Phone)
```
Comment (shimonacarvalho) [inline pkg/handlers/primeapi/mto_agent_test.go:275]: Great checks!

File: pkg/handlers/primeapi/mto_agent_test.go (hunk 1 above)
Comment (abbyoung) [reply pkg/handlers/primeapi/mto_agent_test.go:254]: ohh thank you! i can update

File: pkg/handlers/primeapi/mto_agent_test.go [hunk 3]
```diff
@@ -204,3 +206,170 @@ func (suite *HandlerSuite) TestUpdateMTOAgentHandler() {
...
+			MtoShipmentID: *handlers.FmtUUID(newMTOShipment.ID),
+			Body:          payload,
+		}
+		// Run swagger validations
+		suite.NoError(params.Body.Validate(strfmt.Default))
+
+		// Run handler and check response
+		response := handler.Handle(params)
+		suite.IsType(&mtoshipmentops.CreateMTOAgentUnprocessableEntity{}, response)
+	})
```
Comment (shimonacarvalho) [inline pkg/handlers/primeapi/mto_agent_test.go:374]: Awesome tests, and comments. Helped me review a lot more easily.

File: pkg/handlers/primeapi/mto_agent_test.go (hunk 2 above)
Comment (abbyoung) [reply pkg/handlers/primeapi/mto_agent_test.go:275]: whoops i got em ♻️ turned around in my mind! thanks 😄

**Review by shimonacarvalho — APPROVED** [review]
Some changes requested but none are critical. Good work!

File: pkg/handlers/primeapi/mto_agent_test.go (hunk 1 above)
Comment (shimonacarvalho) [reply pkg/handlers/primeapi/mto_agent_test.go:254]: I can explain some time why that is, if it doesn't make sense. It's too convoluted to type out.

Comment (abbyoung) [reply pkg/handlers/primeapi/mto_agent_test.go]: updated ✅

File: pkg/handlers/primeapi/mto_agent_test.go (hunk 2 above)
Comment (abbyoung) [reply pkg/handlers/primeapi/mto_agent_test.go]: updated ✅

**Pull request merged**